### DISCUSSION TITLE
Possible to coexist send_file() and json object in `return ...`?

### DISCUSSION THREAD

**Discussion by solzard** [pallets/flask#5033] [Q&A] [answered]
I have the following API endpoint that accepts an image to analyze and returns objects found in the image as a collection of cropped images.

```python
@api.post("/detect")
def detect() -> ResponseReturnValue:
    """Detect objects in an image and return each object as a separate image."""
    # Machine learning logic
    ...

    # Example object data
    object_data: dict = {
        "dog_1.jpg": {"label": "dog", "score": 0.99, "box": [0, 0, 100, 100]},
        "dog_2.jpg": {"label": "dog", "score": 0.99, "box": [0, 0, 100, 100]},
        "person_1.jpg": {"label": "person", "score": 0.99, "box": [0, 0, 100, 100]},
    }
    zip_file: Path = Path("/path/to/zipfile")

    return send_file(zip_file)
```

Currently, it only returns a zip file of the collection (`zip_file` variable); however, I want to make it also returns the detailed data of the collection (`object_data` variable).
How can I make it returns both `send(zip_file)` and a json message `{"message": "Success", "objects": object_data}`?

What I tried:
- 1: `return {"message": "Success", "objects": object_data}, send_file(zip_file)` raised `TypeError: Invalid status argument`.
- 2: `return send_file(zip_file), {"message": "Success", "objects": object_data}` 
  - It returned only the zip file.
  - Although `objects` was included in headers, it was `str` not `dict`

Environment:
```toml
python = "^3.11"
flask = "^2.2.3"
torch = "^2.0.0"
torchvision = "^0.15.1"
opencv-python = "^4.7.0.72"
```

**Comment by RonnyPfannschmidt** [comment, marked as answer]
its not quite clear what you are asking for, there can only be one response, itsc currently either the json or the zip file

without providing details on the intended conusmption its unclear how to do what you want

**Reply by solzard** [reply]
> either the json or the zip file

That's the type of answer I needed. I wondered if there was a special way to mix two responses into one that I didn't know. 

It's now clear that no magic exists. I'll embed json file in the zip file, instead. 
Thank you for your quick reply, @RonnyPfannschmidt!

**Comment by davidism** [comment]
You can return any data you want, but you can only return one thing. You can combine multiple files into one block of data, such as by zipping them, but it's up to you to also be able to decode that again on the other side.

**Reply by solzard** [reply]
Can I ask for other possible ways to _combine multiple files into one block of data_ besides zipping?

**Reply by RonnyPfannschmidt** [reply]
commonly any archive format is feasible
